fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/RequestedOrdering.java

```diff
@@ -86,6 +90,10 @@ public boolean isDistinct() {
         return distinctness == Distinctness.DISTINCT;
     }
 
+    public boolean isExhaustive() {
```

Comment:
At least one of these methods or the class itself needs a comment on what `exhaustive` means in this context.

Reply:
I have it on the attribute declaration, will add that here as well.

Reply:
Done.

Reply:
So this enables the property to signal whether the enforcer is interested in either getting the best plan that satisfies that particular property, or enumerate all candidate plans that satisfy it instead. Correct?

Can we think of this as a parameterized branch-and-bound? in other words, can we think of the `exhaustiveness` attribute as a way to apply some sort of costing in-place? If so, would it make sense to think of a more generalized form of costing instead of carrying it out through this boolean attribute?

Reply:
I'd say the exhaustiveness is part of the requirement, it's not a modifier of the requirement. As much as the ordering requirement is already facilitating branch-and-bound (which is currently exhaustive), this is now doing away with that requirement if we can and just requesting the best plan.

There are other things you may want to request (wrt to orderings) that will be more complicated than exhaustiveness. For instance, for joins with a requested ordering of `a, b, c, d` you should create plans on the outer with strict orders for `a`, `a, b`, `a, b, c`, and a regular (non-strict) order of `a, b, c, d`. That's a lot of requested orders. Alternatively you could also say, give me all orders where the order is a prefix of `a, b, c, d` and strict. My point here is that the requested requirement (requested ordering here) is ours to formulate and depends very much on the operators that are in play, etc. This will become much more elaborate over time to capture all the needs we will have from the sub graph.